docs/working_with_splat.md

```diff
@@ -0,0 +1,122 @@
+# Working with Splat
+
+## What is Splat?
+
+The base of this project is the Doubutsu no Mori rom, which is a 25mb chunk of data. We need to split this file into separate files in order to work on them. Because this is copyrighted data it can't be added to the repo, so we need a system where contributors can bring their own rom and extract the files outside of the repo. To accomplish this we use splat. 
+
+Splat only handles file extraction. It generates linker scripts, but it doesn't do any of the building.
+
+## Overview of Configuration Files:
+
+`yamls/jp/header.yaml`
+
+The main configuration file for splat. Most of the time you can ignore this.
+
+`yamls/jp/boot.yaml`  
+`yamls/jp/code.yaml`  
+`yamls/jp/overlays.yaml`
+
+These yaml files are responsible for defining each file from the rom. Currently we don't have a complete definition of every file, so these files are constantly being updated as we learn more.
+
+`animalforest.jp.yaml`
+
+This is an automatically generated file that combines the other yaml files. you shouldn't edit this file, as changes will just be overwritten when it gets generated again.
+
+`linkerscripts/jp/symbol_addrs_boot.txt`  
+`linkerscripts/jp/symbol_addrs_code.txt`  
+`linkerscripts/jp/symbol_addrs_overlays.txt`
+
+By default functions will be named function_XXXXXXXX_jp, data will be named D_XXXXXXXX_jp and bss will be named B_XXXXXXXX_jp. These can be renamed by defining them in these files.
+
+## Starting work on a file (boot or code)
+
+Make sure you have the appropriate baserom.z64 setup. see the [readme](../README.md) for more details.
+
+The first step is to have splat extract everything. This is done with `make extract -j`. Because all the file splits haven't been determined yet, it's possible the file you want to work on is not split correctly.

```

Comment:
We usually try to avoid recommending only passing the `-j` flag to `make` because it means invoking as many jobs as possible, which can result on some PCs hanging, specially the low end PCs.
Instead it should be recommended to pass `-j N`, with N being an integer number that does not exceed the core count of your PC